src/compile/facet.ts

```diff
@@ -184,32 +184,25 @@ export class FacetModel extends Model {
   public parseMark() {
     this.child().parseMark();
 
-    this.component.mark = extend(
-      {
-        name: this.name('cell'),
-        type: 'group',
-        from: extend(
-          {data: this.dataName(SOURCE)},
-          {
-            transform: [{
-              type: 'facet',
-              groupby: [].concat(
-                this.has(ROW) ? [this.field(ROW)] : [],
-                this.has(COLUMN) ? [this.field(COLUMN)] : []
-              )
-            }]
-          }
-        ),
-        properties: {
-          update: getFacetGroupProperties(this)
+    this.component.mark = [{
```

Comment:
I agree that you move `this.child.assembleGroup()` to `assembleMark()`.  But why does this need to become an array?